operators/test/e2e/cmd/run/run.go

```diff
@@ -0,0 +1,404 @@
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to load kubeconfig")
+		}
+
+		clientConfig = clientcmd.NewDefaultClientConfig(*kubeConf, &overrides)
+	} else {
+		loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+		clientConfig = clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, &overrides)
+	}
+
+	// create the kubernetes API client
+	config, err := clientConfig.ClientConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create kubernetes client config")
+	}
+
+	return kubernetes.NewForConfig(config)
+}
+
+// monitorTestJob keeps track of the test pod to determine whether the tests failed or not.
+func (s *helper) monitorTestJob(client *kubernetes.Clientset) error {
+	log.Info("Waiting for test job to start")
+	ctx, cancelFunc := context.WithTimeout(context.Background(), jobTimeout)
+	defer func() {
+		cancelFunc()
+		runtime.HandleCrash()
+	}()
+
+	factory := informers.NewSharedInformerFactoryWithOptions(client, kubePollInterval,
+		informers.WithNamespace(s.testContext.E2ENamespace),
+		informers.WithTweakListOptions(func(opt *metav1.ListOptions) {
+			opt.LabelSelector = fmt.Sprintf("%s=%s", testRunLabel, s.testContext.TestRun)
+		}))
+
+	informer := factory.Core().V1().Pods().Informer()
+
+	jobStarted := false
+	streamStatus := make(chan error, 1)
+	var err error
+
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			pod := obj.(*corev1.Pod)
+			log.Info("Pod added", "name", pod.Name)
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			newPod := newObj.(*corev1.Pod)
+			switch newPod.Status.Phase {
+			case corev1.PodRunning:
+				if !jobStarted {
+					jobStarted = true
+					log.Info("Pod started", "name", newPod.Name)
+					go s.streamTestJobOutput(streamStatus, client, newPod.Name)
+				} else {
+					select {
+					case streamErr := <-streamStatus:
+						if streamErr != nil {
+							log.Error(streamErr, "Stream failure")
+							err = streamErr
+							cancelFunc()
+						}
+					default:
+					}
+				}
+			case corev1.PodSucceeded:
+				log.Info("Tests completed successfully", "name", newPod.Name)
+				cancelFunc()
+			case corev1.PodFailed:
+				log.Info("Pod is in failed state", "name", newPod.Name)
+				err = errors.New("tests failed")
+				cancelFunc()
+			default:
+				log.Info("Waiting for pod to be ready", "name", newPod.Name, "status", newPod.Status.Phase)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			pod := obj.(*corev1.Pod)
+			log.Info("Pod deleted", "name", pod.Name)
+			cancelFunc()
+		},
+	})
+
+	informer.Run(ctx.Done())
+	return err
+}
+
+func (s *helper) streamTestJobOutput(streamStatus chan<- error, client *kubernetes.Clientset, pod string) {
+	log.Info("Streaming pod logs", "name", pod)
+	defer close(streamStatus)
+	sinceSeconds := int64(30)
+	opts := &corev1.PodLogOptions{
+		Container:    "e2e", //TODO remove hardcoded value
+		Follow:       true,
+		SinceSeconds: &sinceSeconds,
+	}
```

Comment:
In this particular instance, waiting for the test pod to launch, figuring out its name and then tailing the logs is hard to do with kubectl because it requires several commands to be executed in a loop and some piping (see https://github.com/elastic/cloud-on-k8s/blob/master/operators/hack/run-e2e.sh). 

I tried to only use kubectl when it was simpler than using the API (eg. installing resources from arbitrary YAML files would require several hundred lines of code if done through the API)